src/client/hud.cpp

```diff
@@ -545,14 +545,21 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 				}
 				break; }
 			case HUD_ELEM_MINIMAP: {
-				if (e->size.X <= 0 || e->size.Y <= 0)
-					break;
 				if (!client->getMinimap())
 					break;
 				// Draw a minimap of size "size"
 				v2s32 dstsize(e->size.X * m_scale_factor,
 				              e->size.Y * m_scale_factor);
-				// (no percent size as minimap would likely be anamorphosed)
+
+				// Only one percentage is supported to avoid distortion.
+				if (e->size.X < 0)
+					dstsize.X = dstsize.Y = m_screensize.X * (e->size.X * -0.01);
+				 else if (e->size.Y < 0)
+					dstsize.X = dstsize.Y = m_screensize.Y * (e->size.Y * -0.01);
+
+				if (dstsize.X <= 0 || dstsize.Y <= 0)
```

Comment:
Can this still happen? It seems me that either we already have `size.X > 0 && size.Y > 0` => doesn't happen (we get the `dstsize` as the product of `size` with something positive then), `size.X < 0 || size.Y < 0` => above code runs, sets it to something positive (because screensize is multiplied with something positive).

Reply:
if `size.X == 0 || size.Y == 0`

Reply:
right, probably replace it with that, would be a bit clearer imo